### PR TITLE
Migrate to yarn

### PR DESCRIPTION
**Pull request by PVince81**
- [x] TEST: `make` still works
- [x] TEST: `make dist` still works
- [x] TEST: ownCloud still runs fine

- [x] Expecting CI failures. Will require `npm install -g yarn` on CI.
- [x] build server require yarn

Do we want to backport this ?

### CODE REVIEW

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/29906

**Comment by PVince81** [comment]
- [ ] dev docs update

**Comment by DeepDiver1975** [comment]
- [x] yarn is installed on jenkins

File: tests/drone/yarn-install.sh [hunk 1]
```diff
@@ -6,4 +6,5 @@ if [[ "$(pwd)" == "$(cd "$(dirname "$0")"; pwd -P)" ]]; then
   exit 1
 fi
 
-exec npm install --prefix build
+cd build
+exec yarn install
```
Comment (tboerger) [inline tests/drone/yarn-install.sh]: maybe `exec yarn install --modules-folder build/node_modules` won't require any dir change like before with npm?

Comment (DeepDiver1975) [reply tests/drone/yarn-install.sh]: Tried that - did not work out

Comment (PVince81) [reply tests/drone/yarn-install.sh]: indeed, I noticed that yarn doesn't have the `--prefix` argument but also doesn't complain if you use it... see makefile

Comment (DeepDiver1975) [reply tests/drone/yarn-install.sh]: yarn install --cwd build

**Comment by PVince81** [comment]
:+1: for @DeepDiver1975's changes.

Need review for mine.

@felixheidecke if you're familiar with yarn you could check them 😄

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**